pipelines/common/src/test/java/com/google/fhir/analytics/view/SQLonFHIRv2Test.java

```diff
@@ -0,0 +1,241 @@
+          "fn_boundary.date lowBoundary",
+          "fn_boundary.date highBoundary",
+
+          // TODO the error condition here does not seem right.
+          "validate.wrong type in forEach",
+
+          // Reason: getResourceKey is used inside the FHIRPath which we don't support yet.
+          // TODO: Implement the above feature.
+          "fn_reference_keys.getReferenceKey result matches getResourceKey without type specifier",
+          "fn_reference_keys.getReferenceKey result matches getResourceKey with right type"
+              + " specifier",
+          "fn_reference_keys.getReferenceKey result matches getResourceKey with wrong type"
+              + " specifier"
+      );
+
+  @Test
+  public void runAllTests() throws IOException {
+    // TODO make the FHIR version optional.
+    IParser parser = FhirContext.forR4Cached().newJsonParser();
+    String testsRoot = Resources.getResource("sql-on-fhir-v2-tests").getPath();
+    Path testsPath = Paths.get(testsRoot);
+    List<Path> testFiles =
+        Files.walk(testsPath)
+            .filter(f -> f.getFileName().toString().endsWith(".json"))
+            .collect(Collectors.toList());
+    for (Path p : testFiles) {
+      InputStream stream = new FileInputStream(p.toFile());
+      String jsonContent = IOUtils.toString(stream, StandardCharsets.UTF_8);
+      JSONObject json = new JSONObject(jsonContent);
+      String collectionTitle = (String) json.get("title");
+      log.info("Next test-collection: " + collectionTitle);
+      List<IBaseResource> resources = new ArrayList<>();
+      JSONArray resourceArray = (JSONArray) json.get("resources");
+      for (Object r : resourceArray) {
+        resources.add(parser.parseResource(r.toString()));
+      }
+      for (Object t : (JSONArray) json.get("tests")) {
+        JSONObject testObj = (JSONObject) t;
+        String testTitle = (String) testObj.get("title");
+        if (SKIPPED_TESTS.contains(collectionTitle + "." + testTitle)) continue;
+        // Note: To debug a single test case we can do the following:
+        // if (!testTitle.equals("wrong type in forEach")) continue;
+        log.info("Next test: " + testTitle);
+        ExpectedRows expectedRows = null; // will be null if `expectError` is set.
+        if (!testObj.has("expectError")) {
+          expectedRows = new ExpectedRows((JSONArray) testObj.get("expect"));
+        }
+        try {
+          ViewDefinition view =
+              ViewDefinition.createFromString(testObj.get("view").toString(), false);
+          ViewApplicator applicator = new ViewApplicator(view);
+          int totalRows = 0;
+          for (IBaseResource resource : resources) {
+            if (!view.getResource().equals(resource.fhirType())) continue;
+            RowList rowList = applicator.apply(resource);
+            for (FlatRow row : rowList.getRows()) {
+              assertThat("Row not found; index " + totalRows, expectedRows.hasRow(row));
```

Comment:
Not sure if the `index` can give some information here for debugging, as the actualRow position is not checked exactly in the expectedRows.

Reply:
`totalRows` is actually being increased inside this `for` loop; so the current `totalRows` is the actual row on which the assertion has failed.